ports/stm32/mpconfigport.h

```diff
@@ -271,12 +271,24 @@ static inline mp_uint_t disable_irq(void) {
 
 #define MICROPY_THREAD_YIELD() pyb_thread_yield()
 #else
+#if defined(STM32L1)
 #define MICROPY_EVENT_POLL_HOOK \
     do { \
+        mp_uint_t irq_state; \
         extern void mp_handle_pending(bool); \
         mp_handle_pending(true); \
+        irq_state = MICROPY_BEGIN_ATOMIC_SECTION(); \
```

Comment:
why does it take the atomic state?

Reply:
This is the same reason with modmachine.c.